services/http/src/main/java/org/mule/services/http/impl/service/server/grizzly/GrizzlyServerManager.java

```diff
@@ -255,11 +250,43 @@ private int retrieveMaximumHeaderSectionSize() {
     try {
       return valueOf(getProperty(MAXIMUM_HEADER_SECTION_SIZE_PROPERTY_KEY, String.valueOf(DEFAULT_MAX_HTTP_PACKET_HEADER_SIZE)));
     } catch (NumberFormatException e) {
-      throw new MuleRuntimeException(CoreMessages.createStaticMessage(String.format("Invalid value %s for %s configuration",
-                                                                                    getProperty(MAXIMUM_HEADER_SECTION_SIZE_PROPERTY_KEY),
-                                                                                    MAXIMUM_HEADER_SECTION_SIZE_PROPERTY_KEY)),
+      throw new MuleRuntimeException(createStaticMessage(String.format("Invalid value %s for %s configuration",
+                                                                       getProperty(MAXIMUM_HEADER_SECTION_SIZE_PROPERTY_KEY),
+                                                                       MAXIMUM_HEADER_SECTION_SIZE_PROPERTY_KEY)),
                                      e);
     }
   }
 
+  /**
+   * Wrapper that adds startup and disposal of manager specific data.
+   */
+  private class GrizzlyHttpServerWrapper extends GrizzlyHttpServer {
+
+    //TODO - MULE-11117: Cleanup GrizzlyServerManager server specific data
+
+    public GrizzlyHttpServerWrapper(ServerAddress serverAddress, TCPNIOTransport transport,
+                                    HttpListenerRegistry listenerRegistry, Supplier<Scheduler> schedulerSupplier) {
+      super(serverAddress, transport, listenerRegistry, schedulerSupplier);
+    }
+
+    @Override
+    public synchronized void start() throws IOException {
+      idleExecutorPerServerAddressMap.get(this.getServerAddress()).start();
+      super.start();
+    }
+
+    @Override
+    public synchronized void dispose() {
+      super.dispose();
```

Comment:
it's done below in two parts, first doing the dispose and then removing the reference.